src/lib.rs

```diff
@@ -140,21 +139,94 @@ impl Wiki {
 
         Ok(())
     }
+    /// read all the files from current paths
+    pub fn read_files(&self, file_directory: &str) {
+        //create the default file path
+        if !Path::new(file_directory).exists(){
+            info!("Creating directory for file input: {}.", file_directory);
+            fs::create_dir(file_directory);
+        }
+        for path in self.paths.clone(){
+            let mut count = 0;
+
+            //create the path for the html site or just find it
+            let page_files_path = format!("{}/{}", file_directory, path.to_str().unwrap()).replace(".md","");
+            if !Path::new(page_files_path.as_str()).exists(){
+                info!("Creating directory for {}'s files: {}.", path.to_str().unwrap(), page_files_path.as_str());
+                fs::create_dir_all(page_files_path.as_str()).unwrap();
+            }
+
+            //add the path
+            let file_path = Path::new(page_files_path.as_str());
+            //read all the files
+            let files = file_path.read_dir().unwrap();
+
+            //now attach the files to the html sites folder
+            for entry in files {
+                let current_entry = match entry {
+                    Ok(entry) => entry,
+                    _ => panic!("Couldn't read the path!"),
+                };
+
+                let link = format!(
+                    "\n<a href='http://localhost:3000/files/{}/{}'>{}</a><br>\n",
+                    path.to_str().unwrap().replace(".md", ""),
+                    current_entry.file_name().into_string().unwrap().replace("\n","."),
+                    current_entry.file_name().into_string().unwrap()
+                );
+
+                let mut html_file = OpenOptions::new().read(true).write(true).create(true).
+                    open(format!("output/{}", path.to_str().unwrap().
+                    replace(".md", ".html"))).unwrap();
+
+                let mut buffer = String::new();
+
+                html_file.read_to_string(&mut buffer);
+                buffer += link.clone().as_str();
+                info!("Creating link to {:?} for {:?}", current_entry.file_name(), path.to_str().unwrap().replace(".md", ".html"));
+                html_file.write((&buffer).as_bytes());
+                count += 1;
+            }
+            //check if there are no files attached
+            if count == 0 {
+                info!("No files found so far for {}. Simply add your files to the folders in 'files'.", path.to_str().unwrap())
+            }
+        }
+    }
 
     /// Create an HTTP server serving the generated files
     pub fn serve(&self, output_directory: &str) -> Result<()> {
         // Create a default listening address
-        let addr = "localhost:5000";
+        let addr = "localhost:3000";
```

Comment:
I recommend to use a higher ranged port here. The address should be a static string somewhere in the lib, since we have no configuration options.